hedera-node/src/test/java/com/hedera/services/contracts/execution/HederaTracerTest.java

```diff
@@ -119,7 +119,6 @@ void initializesActionsAsExpectedOnNewFrames() {
         final var firstChildFrame = mock(MessageFrame.class);
         given(firstChildFrame.getType()).willReturn(Type.CONTRACT_CREATION);
         given(firstChildFrame.getCode()).willReturn(Code.EMPTY);
-        given(firstChildFrame.getSenderAddress()).willReturn(contract);
```

Comment:
SHouldn't the test change with the code change.
We should have an explicit check for `parentFrame` not `nextFrame`

Reply:
Added a couple of `verify` statements on the number of calls to `getContractAddress` for the `topLevelFrame` (`parentFrame`). 

We also have pre-existing asserts (although maybe implicit) that the [`callingContract` for the first child frame ](https://github.com/hashgraph/hedera-services/blob/bf42220db0bb417cd77a4181169357751ad05ce9/hedera-node/src/test/java/com/hedera/services/contracts/execution/HederaTracerTest.java#L141) is the [`contractAddress` of the top-level frame.](https://github.com/hashgraph/hedera-services/blob/bf42220db0bb417cd77a4181169357751ad05ce9/hedera-node/src/test/java/com/hedera/services/contracts/execution/HederaTracerTest.java#L92)